pkg/devfile/adapters/common/command_test.go

```diff
@@ -10,215 +10,126 @@ import (
-				},
+			name: "Case 3: Valid devfile with devinit and devrun",
+			execCommands: []versionsCommon.Exec{
+				getExecCommand("", initGroup),
+				getExecCommand("", runGroup),
 			},
-			isCommandRequired: []bool{false},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{initGroup, runGroup},
+			wantErr:       false,
 		},
 		{
-			name:              "Case 8: Invalid devfile with wrong devbuild command type",
-			requestedCommands: []string{"devbuild"},
-			commandActions: []common.DevfileCommandAction{
+			name: "Case 4: Invalid devfile with empty component",
+			execCommands: []versionsCommon.Exec{
 				{
-					Command:   &commands[0],
-					Component: &components[0],
-					Workdir:   &workDir[0],
-					Type:      &invalidCommandType,
+					CommandLine: commands[0],
+					Component:   emptyString,
+					WorkingDir:  workDir[0],
+					Group:       &versionsCommon.Group{Kind: initGroup},
 				},
 			},
-			isCommandRequired: []bool{true},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{initGroup},
+			wantErr:       true,
 		},
 		{
-			name:              "Case 9: Invalid devfile with empty devbuild component",
-			requestedCommands: []string{"devbuild"},
-			commandActions: []common.DevfileCommandAction{
+			name: "Case 5: Invalid devfile with empty devinit command",
+			execCommands: []versionsCommon.Exec{
 				{
-					Command:   &commands[0],
-					Component: &emptyString,
-					Workdir:   &workDir[0],
-					Type:      &validCommandType,
+					CommandLine: emptyString,
+					Component:   components[0],
+					WorkingDir:  workDir[0],
+					Group:       &versionsCommon.Group{Kind: initGroup},
 				},
 			},
-			isCommandRequired: []bool{false},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{initGroup},
+			wantErr:       true,
 		},
 		{
-			name:              "Case 10: Invalid devfile with empty devbuild command",
-			requestedCommands: []string{"devbuild"},
-			commandActions: []common.DevfileCommandAction{
+			name: "Case 6: Valid devfile with empty workdir",
+			execCommands: []common.Exec{
 				{
-					Command:   &emptyString,
-					Component: &components[0],
-					Workdir:   &workDir[0],
-					Type:      &validCommandType,
+					CommandLine: commands[0],
+					Component:   components[0],
+					Group:       &versionsCommon.Group{Kind: runGroup},
 				},
 			},
-			isCommandRequired: []bool{false},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{runGroup},
+			wantErr:       false,
 		},
 		{
-			name:              "Case 11: Valid devfile with empty workdir",
-			requestedCommands: []string{"devrun"},
-			commandActions: []common.DevfileCommandAction{
+			name: "Case 7: Invalid command referencing an absent component",
+			execCommands: []common.Exec{
 				{
-					Command:   &commands[0],
-					Component: &components[0],
-					Type:      &validCommandType,
+					CommandLine: commands[0],
```

Comment:
a test case where multiple commands of a type exist but the default is returned

Reply:
Added